src/test/java/junit/org/rapidpm/frp/model/QuadTest.java

```diff
@@ -0,0 +1,110 @@
+package junit.org.rapidpm.frp.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+import org.rapidpm.frp.model.Quad;
+
+public class QuadTest {
+
+  @Test
+  public void testEquals() {
+    final Quad quad = new Quad(0, 1, 2, 3);
+
+    assertTrue(quad.equals(quad));
+    assertFalse(quad.equals(null));
+    assertFalse(quad.equals(new Quad(null, 1, 2, 3)));
+    assertFalse(quad.equals(new Quad(0, null, 2, 3)));
+    assertFalse(quad.equals(new Quad(0, 1, null, 3)));
+  }
+
+  @Test
+  public void testEqualsT1() {
+    final Quad quad = new Quad(null, 1, 2, 3);
```

Comment:
maybe call these tests `testEqualsT1Null` to make clear that it's the position of the `null` you're changing?

Reply:
I disagree